### ISSUE TITLE
Access violation

### ISSUE THREAD

**Issue by idnovic** [bgrabitmap/lazpaint#96]
Platform: Windows 10

What happens:
I open lazpaint
I am presented with an dialogue with an error message. (see attachment)

What did I expect:
I open lazpaint
no error message

<img width="275" alt="lazpaint error" src="https://user-images.githubusercontent.com/28260793/69278058-d1956f00-0be1-11ea-8390-86013f18eb2b.png">

**Comment by idnovic** [comment]
I can attach a log file if someone tells me where to find it

**Comment by lainz** [comment]
Which version of LazPaint are you running?
Is from the installer or a portable zip file?
What is the bitness of your OS?
If it's a portable file, from which path are you running it?
Try running it as administrator to see if because it can't write a config file.
Or try disabling your antivirus / firewall since it connects to internet to check if a new version is available.

**Comment by idnovic** [comment]
I am running lazpaint 7.0.7
the installer version not portable
64 bit windows 10 1909
I tried as admin, same error

I am not running any antivir/firewall. Even disabled windows defender.

Can I somehow check why the error message is presented?
A log would help me.

**Comment by lainz** [comment]
AFAIK there is no log file.

You must compile the application in debug mode and it hopefully will throw the line where the error is.

To compile: Install Lazarus 2.0.6, install bgrabitmap, bgracontrols and lazpaintcontrols. Change the build mode to debug, and debug it.

**Comment by circular17** [comment]
Hmmm I suppose i can add a global error handler to display more information.

**Comment by lainz** [comment]
> Hmmm I suppose i can add a global error handler to display more information.

https://github.com/r3code/lazarus-exception-logger

**Comment by idnovic** [comment]
That would be awesome. I did not find time to compile it myself. I did look into debugging on windows but still need to test it.
I found an attachable process debugger.

**Comment by JohnTHaller** [comment]
It may have something to do with having a specific debugger or graphics driver installed. I can recreate this error using both the 32-bit and 64-bit zipped versions as well as the 64-bit installed version on my Windows 10 x64 main machine with build 1903. It does not occur on a clean Windows 10 x64 virtual machine in VirtualBox. My primary machine has multiple versions of Delphi installed and is using the current Nvidia drivers for a GTX 1060.

**Comment by circular17** [comment]
Hmm just a guess, it could be related to the user name containing special characters, thus making an error in the configuration directory.

**Comment by idnovic** [comment]
> Hmm just a guess, it could be related to the user name containing special characters, thus making an error in the configuration directory. Otherwise, I don't know, I could try and go through the code at startup. But really debugging is the best way to find.

My username does not contain special characters.
I did not find time yet to debug the problem. I hope I will find time this weekend.

**Comment by circular17** [comment]
Hi there.

I was not able to compile ExceptionLogger on Linux. 

So I've added on the dev branch some basic dialog when an unhandled exception occurs.

**Comment by lainz** [comment]
Hi, this is the Windows 64 bit binary from today sources
[lazpaint-debug-win64-2020-02-04.zip](https://github.com/bgrabitmap/lazpaint/files/4156796/lazpaint-debug-win64-2020-02-04.zip)

**Comment by circular17** [comment]
Thank you Lainz

**Comment by lainz** [comment]
You're welcome. And anyone tested the build with debug information?

**Comment by lainz** [comment]
Tested lazpaint 32 on macOS with "Wine is not an emulator" and works fine, thats another option at least :)

**Comment by circular17** [comment]
Cool. That's indeed a good way to bypass some problems with MacOS compatibility.

**Comment by idnovic** [comment]
> You're welcome. And anyone tested the build with debug information?

I reinstalled windows. I am testing tonight but I can not compare to the previous system config.

**Comment by circular17** [comment]
Ok, well if it works that's good news as well.

**Comment by idnovic** [comment]
works without issue atm.

**Comment by circular17** [comment]
We will see with next test version of LazPaint what happens.

**Comment by JohnTHaller** [comment]
This is still occurring for me with the 7.1.1 release on my Windows 10 64-bit machine with both the 32-bit and 64-bit builds whether installed or not.

64-bit error: Error initializing application!

Exception class: EAccessViolation
Message: Access violation
Stacktrace:
    $0000000100554BF8
    $000000010055429A
    $0000000100554250
    $000000010005B0E8
    $0000000100020E0E
    $000000010001F09E
    $000000010002668F
    $000000010002DEC4
    $000000010004E710
    $0000000100051ABB
    $0000000100003A85
    $0000000100003AE6
    $00000001000183E0
    $0000000100003157
    $00007FF8555B7BD4
    $00007FF856DCCED1

Application will now close.

32-bit Error: Error initializing application!

Exception class: EAccessViolation
Message: Access violation
Stacktrace:
    $00898003
    $00897899
    $0044E751
    $0041CED9
    $0041B624
    $00421A86
    $00427E1B
    $00443E74

Application will now close.

**Comment by circular17** [comment]
Ok can you try with the debug version 7.0.9?

**Comment by JohnTHaller** [comment]
> Ok can you try with the debug version 7.0.9?

Is there a way to direct the output as opposed to trying to copy and paste the dozen+ messagebox windows that are created?

**Comment by circular17** [comment]
Nope, but just the first one will be useful.

**Comment by JohnTHaller** [comment]
Here's the first screen test and I attached the first 1,000+ lines as a TXT file.
```
Exception class: EAccessViolation
Message: Access violation
Stacktrace:
    $00000001005D1B41  CLOSE,  line 467 of tablet/Tablet.pas
    $00000001005D0CFA  DESTROY,  line 121 of tablet/laztabletwin.pas
    $00000001005D0C67  CREATE,  line 117 of tablet/laztabletwin.pas
    $00000001000696F0  FORMCREATE,  line 950 of lazpaintmainform.pas
    $00000001000273AE  DOCREATE,  line 896 of include/customform.inc
    $000000010002563E  AFTERCONSTRUCTION,  line 77 of include/customform.inc
    $000000010002CC2F  CREATE,  line 3135 of include/customform.inc
    $0000000100034464  CREATEFORM,  line 2241 of include/application.inc
    $000000010005720C  FORMSNEEDED,  line 439 of lazpaintinstance.pas
    $000000010005BF81  SHOW,  line 1366 of lazpaintinstance.pas
    $0000000100003D93  main,  line 197 of lazpaint.lpr
    $0000000100003E36  main,  line 208 of lazpaint.lpr
    $00000001000190B0
    $0000000100003177
    $00007FFE56AB7BD4
    $00007FFE589ACED1

Application will now close.
```
[LaxPaint709Debug.txt](https://github.com/bgrabitmap/lazpaint/files/4510297/LaxPaint709Debug.txt)

**Comment by circular17** [comment]
Thanks very much. So it is a problem with the initialization of the tablet. I will add some code to avoid the crash. In this case, the tablet "pressure" would not be available.

I will provide a debug version to test it.

**Comment by idnovic** [comment]
I wanted to add that I did run lazpaint on a surface go. So it maybe was the same error for me

**Comment by circular17** [comment]
Hmm you're not using this computer anymore then so you cannot reproduce the bug?

**Comment by idnovic** [comment]
I do. Even with a pen but lazpaint does not crash for me.
I am using lazpaint 7.1.1.

**Comment by circular17** [comment]
Ok so I consider that issue to be fixed.

**Comment by JohnTHaller** [comment]
7.1.1 still crashes for me, both 32-bit and 64-bit, as per my earlier comment: https://github.com/bgrabitmap/lazpaint/issues/96#issuecomment-612633379

The PC it is crashing on is a desktop without a touch screen. It used to have a USB drawing tablet connected which it no longer has.

**Comment by circular17** [comment]
When I say fixed I mean fixed on dev branch and next release.

**Comment by JohnTHaller** [comment]
> When I say fixed I mean fixed on dev branch and next release.

Ah, thanks. I didn't see a place to download the dev test. I can confirm this is fixed in 7.1.2. Thanks!
